### DISCUSSION TITLE
Plebian on pine64 Quartz64 B

### DISCUSSION THREAD

**Discussion by Eve1374** [Plebian-Linux/plebian-website#4] [General]
Hi !
I received a Quartz64 single board computer, and decided to install Plebian.
At first I didn't manage to boot on the bootable micro SD card I thought I made, so I booted with a manjaro image instead and flashed the emmc module with the plebian image, using the command in plebian installation guide : 
`xzcat plebian-debian-bookworm-quartz64b.img.xz | sudo dd of=/dev/sdX bs=16M oflag=dsync status=progress`

it didn't work either (no sign of life when booting on the emmc), so I tried again by extracting the img file manually and ran
`sudo dd bs=4M if=/path/to/plebian.img of=/dev/mmcblk[number] status=progress oflag=sync` 

so that I would re-flash the emmc. After that, it booted ! (Maybe the command on plebian guide is wrong ? should bs be 4M instead of 16M ? ...)

I'm glad it worked and I am tinkering with this now (and inaugurating Plebian's forum too). I'm not a highly experienced linux user but managed to configure ethernet connection so I could go on and install the packages I want. 

Looking forward to hear about other users experience !

Eve

**Comment by CounterPillow** [comment]
Hi,

`bs=4M` vs. `bs=16M` shouldn't matter. Are you sure you ran with the right `of=` target device?

**Reply by Eve1374** [reply]
Yes, I am sure. I tried with two different sd cards too.

**Reply by CounterPillow** [reply]
Interesting. The command works for me; it's what I use when flashing images.

**Comment by ThothK** [comment]
Hi Eve1374,

What do you think of the Quartz64 Model B so far?

How did you get ethernet up and running? I have not been able to get ethernet up and running?

Also, did you notice a trouble signal via the LED after two or three days? The Q64B I have running Plebian will emit a trouble code by blinking the LED at a high rate. Restarting it seems to reset the problem for me.

Currently I connect to the Q64B via the wireless interface and would rather connect via the ethernet interface.


-thoth-

**Reply by CounterPillow** [reply]
If Ethernet is not working out of the box for you on your Model B then Pine Store shipped you a broken board. I know this because they've done the same to me, and another person previously.

**Reply by Eve1374** [reply]
Hi, Ethernet was up and running out of the box. I didn't notice anything on the LEDs.
Except that I can't really tell "what I think" because unfortunately I left it aside for a few weeks. My goal is to make a VPN access on this computer but I don't really have knowledge on this subject, so I have to read and try more.
Eve

**Comment by ThothK** [comment]
Hi CounterPillow,

Ugh. It almost seems like I received a number of bad boards. Unfortunately I am well outside of the return window.

I unplugged everything, waited about ten seconds, and then reconnected everything. 
There was a lot of ethernet port LED activity, so much that I thought it was going to work.
Then the ethernet LED started blinking once about every three seconds.

nmtui:
Could not activate connection: 
     Activation failed: The device could
     not be readied for configuration 


DMesg:
[    5.739822] rk_gmac-dwmac fe010000.ethernet: IRQ eth_lpi not found
...
[    5.741635] rk_gmac-dwmac fe010000.ethernet: phy regulator is not available yet, deferred probing
...
[   29.430111] rk_gmac-dwmac fe010000.ethernet end0: Register MEM_TYPE_PAGE_POOL RxQ-0
[   29.432291] rk_gmac-dwmac fe010000.ethernet end0: validation of  with support 00000000,00000000,00006000 and advertisement 00000000,00000000,00000000 failed: -EINVAL
[   29.433942] rk_gmac-dwmac fe010000.ethernet end0: __stmmac_open: Cannot attach to PHY (error: -22)
...
[   30.065007] rk_gmac-dwmac fe010000.ethernet end0: Register MEM_TYPE_PAGE_POOL RxQ-0
... sequence repeats four times ... 
[   30.147359] rk_gmac-dwmac fe010000.ethernet end0: __stmmac_open: Cannot attach to PHY (error: -22)


*sigh*


Thank you for your efforts, CounterPillow. Both here and IRC.


-thoth-

**Reply by CounterPillow** [reply]
If it's multiple bad boards I'm starting to get suspicious. Either that must've been a bad production batch or they did a switcheroo on the PHY chip from out under us. I'll need to look into this further

**Comment by ThothK** [comment]
Hi All,

I received a Quartz64B last week. Yesterday, I installed the eMMC from the first one on it and connected it to power and ethernet.
After boot I could see it in the router's interface and connect to it's ethernet IP via SSH.
It just worked, so yay.

@CounterPillow, it looks like it was the hardware. Again, thank you for your efforts. 


-thoth-

**Comment by beepyaus** [comment]
Just to add my 2 cents. I bought the Quartz-B v.1.3 and there is ethernet issue (I added a post on the pine64 forum) . The Ethernet does not work . I dont know if it's a firmware issue or faulty hardware. ( issue happens on DietPi and Armbian too) 
The LEDs only flash for a few seconds in the boot process , then stop.  
- I have and can use the PoE adapter for the power atleast, so the wires are working on some level. So maybe a driver issue, I just dont have the time or knowledge to know. ...until a later version of a Quartz64 distro is available for testing
-Wifi works, 
- USB 3, connecting a RTL8153 ethernet emits a tonne of kernel warning.

**Reply by beepyaus** [reply]
...the same ethernet dongle does work on USB2

**Reply by CounterPillow** [reply]
There is no firmware for ethernet and the GMAC driver works fine, it's likely to be a hardware issue.

**Comment by GreenGusher** [comment]
I have an ethernet issue too.  I think I can return it.
That issue aside, I'm liking Plebian so far.  I've had only two images work, Manjaro, and Plebian on an sdcard.
